test/lib/checkevents.go

```diff
@@ -0,0 +1,243 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+        https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lib
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"knative.dev/pkg/test/logging"
+)
+
+const (
+	// The interval and timeout used for checking events
+	minEvRetryInterval = 4 * time.Second
+	timeoutEvRetry     = 4 * time.Minute
+)
+
+// Stateful store of events received by the recordevents pod it is pointed at.
+// This pulls events from the pod during any Find or Wait call, storing them
+// locally and triming them from the remote pod store.
+type EventInfoStore struct {
+	getter eventGetterInterface
+
+	lock          sync.Mutex
+	allEvents     []EventInfo
+	firstID       int
+	closeCh       chan struct{}
+	doRefresh     chan chan error
+	timeout       time.Duration
+	retryInterval time.Duration
+}
+
+// Functions used for getting data from the REST api of the recordevents pod.
+// The interface exists for use with unit tests of this module.
+type eventGetterInterface interface {
```

Comment:
nit: do we need Interface as part of the name? I don't have a strong opinion though

Reply:
Probably not.  Given it's purely internal use for unit testing not sure i'll bother changing it unless there are strong opinions, but agreed probably slightly better without.